Chummer/sheets/Notes set.xslt

```diff
@@ -238,6 +215,191 @@
+            <table class="tablestyle">
+              <xsl:call-template name="MartialArtsList"/>
+            </table>
+          </div>
+          <xsl:call-template name="RowSummary">
+            <xsl:with-param name="text" select="$lang.MartialArts"/>
+            <xsl:with-param name="blockname" select="'MartialArtsBlock'"/>
+          </xsl:call-template>
+        </xsl:if>
+
+        <xsl:variable name="cyberlist" select="cyberwares/cyberware[notes != ''] or cyberwares/cyberware/children/cyberware[notes != '']"/>
+        <xsl:if test="$cyberlist">
+          <div class="block" id="CyberwareBlock">
+            <table><tr><td/></tr></table>
+            <xsl:call-template name="TableTitle">
+              <xsl:with-param name="name" 
+                    select="concat($lang.Cyberware,'/',$lang.Bioware)"/>
+            </xsl:call-template>
+            <table class="tablestyle">
+              <tr>
+                <th width="90%" style="text-align: left">
+                  <xsl:value-of select="$lang.Implant"/>
+                </th>
+                <th width="10%"/>
+              </tr>
+              <xsl:for-each select="cyberwares/cyberware[notes != '' or cyberwares/cyberware/children/cyberware/notes != '']">
+                <xsl:call-template name="cybernotes"/>
+              </xsl:for-each>
+            </table>
+          </div>
+          <xsl:call-template name="RowSummary">
+            <xsl:with-param name="text" select="concat($lang.Cyberware,'/',$lang.Bioware)"/>
+            <xsl:with-param name="blockname" select="'CyberwareBlock'"/>
+          </xsl:call-template>
+        </xsl:if>
+
+        <xsl:variable name="armorlist" select="armors/armor[notes != ''] or armors/armor/armormods/armormod[notes != '']"/>
+        <xsl:if test="$armorlist">
+          <div class="block" id="ArmorBlock">
+            <table><tr><td/></tr></table>
+            <xsl:call-template name="TableTitle">
+              <xsl:with-param name="name" select="$lang.Armor"/>
+            </xsl:call-template>
+            <table class="tablestyle">
+              <tr>
+                <th width="90%" style="text-align: left">
+                  <xsl:value-of select="$lang.Armor"/>
+                </th>
+                <th width="10%"/>
+              </tr>
+              <xsl:for-each select="armors/armor[notes != '' or armors/armor/armormods/armormod/notes != '']">
+                <xsl:sort select="name"/>
+                <xsl:call-template name="armornotes"/>
+              </xsl:for-each>
+            </table>
+          </div>
+          <xsl:call-template name="RowSummary">
+            <xsl:with-param name="text" select="$lang.Armor"/>
+            <xsl:with-param name="blockname" select="'ArmorBlock'"/>
+          </xsl:call-template>
+        </xsl:if>
+
+        <xsl:variable name="rangedlist" select="weapons/weapon[type = 'Ranged' and (notes != '' or accessories/accessory/notes != '' or mods/weaponmod/notes != '')]"/>
```

Comment:
Keep in mind, weapons can theoretically nest an infinite number of times via underbarrels. You want a selection that gets every single descendant, not just the first children.